### DISCUSSION TITLE
How to use middleware: bucketEndpointMiddleware

### DISCUSSION THREAD

**Discussion by CN91** [aws/aws-sdk-js-v3#4322] [Q&A]
I am currently in the middle of trying to switch from the old `@aws-sdk/client-s3-node` to the new `@aws-sdk/client-s3`

However in the old situation we used the following config:

```js
      const S3 = new S3Client({
        credentials: {
          accessKeyId: access_key_id,
          secretAccessKey: secret_access_key,
        },
        region: region,
        endpoint: endpoint,
        bucketEndpoint: false,
        forcePathStyle: pathStyleAccess,
      });
```

After removing the old package, and introducing the new one I ran into some problems. 

I can't seem to find an example of how to use the new `bucketEndpointMiddleware` and how to set the property `bucketEndpoint` to false. In almost every thing I can find either the old config is used, and I can't find any documentation on how to actually use the new middleware for this. 

My vscode is telling me all the options I am trying are not correct, due to typings not being correct. 

So can somebody here please provide an example?

**Comment by supersime** [comment]
I've been blocked on this too @CN91 ... I raised a documentation issue back in Feb 2023 (https://github.com/aws/aws-sdk-js-v3/issues/4394) related to the `bucketEndpoint` config value which is as-yet unresolved.

It's now got even worse as the v3 SDK documentation has now moved and yet the v2 to v3 upgrade guide now links to broken pages.

I'm still keen to implement bucketEndpointMiddleware too and can't find a way to do it. Anyone got any ideas - or did you get this resolved?
